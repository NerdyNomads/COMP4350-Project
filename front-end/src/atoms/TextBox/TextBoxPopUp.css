
.TextBoxPopUp-background{

  background-color: rgba(0, 0, 0, 0.692);
  width: 100%;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 50;

  display: flex;
  flex-direction: column;
  align-items: center;

}


.TextBoxPopUp{
    min-width: 300px;
    max-width: 1022px;
    max-height: 693px;
    height:fit-content;

    background: #FFFFFF;
    box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.23);
    border-radius: 15px;

    display: flex;
    flex-direction: column;
    padding: 15px;
    margin: 0px 50px;

    position: fixed;
    z-index: 51;
    top:10%;
}

.TextBoxPopUp-exit{
    align-self: flex-end;
    margin-right: 5px;
    cursor: pointer;
}

.TextBoxPopUp-source{
    margin-left: 20px;
    display: flex;
    flex-direction: row;
    cursor: pointer;
}

.TextBoxPopUp-source-icon{
    margin-right: 8px;

}
.TextBoxPopUp-source-text{
    font-size: 12px;
    line-height: 150%;
    color: #808080;
    align-self: center;
    font-style: italic;
}

.TextBoxPopUp-source-text-a{
  all: unset;
}

.TextBoxPopUp-text{
    text-align: left;
    margin: 20px;
    margin-top: 5px;
    overflow: auto;
}


/* custom scrollbar */
::-webkit-scrollbar {
    width: 20px;
  }
  
  ::-webkit-scrollbar-track {
    background-color: transparent;
  }
  
  ::-webkit-scrollbar-thumb {
    background-color: #d6dee1;
    border-radius: 20px;
    border: 6px solid transparent;
    background-clip: content-box;
  }
  
  ::-webkit-scrollbar-thumb:hover {
    background-color: #a8bbbf;
  }
  /* custom scrollbar ends */