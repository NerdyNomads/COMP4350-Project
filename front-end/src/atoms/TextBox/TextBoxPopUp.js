import React from "react";
import PropTypes from "prop-types";

import { ChainIconSmallIcon, ExitIcon } from "../icons";

import "./TextBoxPopUp.css";

const MAX_CHAR_FOR_SOURCE_TO_SHOW = 50;

function TextBoxPopUp( {text, source, onChangeVisibility} ) {

  const handleBackgroundClick = (e) => {
    // clicked outside of modal
    if(e.target.className == "TextBoxPopUp-background") {
      onChangeVisibility(false);
    }
  };

  const handleExitClick = () => {
    onChangeVisibility(false);
  };

  const handleOnSourceClick = () => {
    alert(`Link copied: ${source}`);

    //Create a temporary element to be able to copy the source
    var temp = document.createElement("textarea");
    temp.value = source;
    document.body.appendChild(temp);
    temp.select();
    document.execCommand("copy");
    document.body.removeChild(temp);
  };

  const renderSource = (srcText) => {
    return srcText.substring(0, MAX_CHAR_FOR_SOURCE_TO_SHOW) + "...";
  };

  return (
    <div className="TextBoxPopUp-background" onClick={handleBackgroundClick}>
      <div className="TextBoxPopUp">
        <div className="TextBoxPopUp-exit" onClick={handleExitClick}>
          <ExitIcon/>
        </div>
        <div className="TextBoxPopUp-source" onClick={handleOnSourceClick}>
          <div className="TextBoxPopUp-source-icon">
            <ChainIconSmallIcon/>
          </div>
          <div className="TextBoxPopUp-source-text">
            <a className="TextBoxPopUp-source-text-a">{renderSource(source)}</a>
          </div>
        </div>
        <div className="TextBoxPopUp-text">
          {text}
        </div>
      </div>
    </div>
  );
}

TextBoxPopUp.propTypes = {
  text: PropTypes.string.isRequired,
  source: PropTypes.string.isRequired,
  onChangeVisibility: PropTypes.func.isRequired,
};


export default TextBoxPopUp;