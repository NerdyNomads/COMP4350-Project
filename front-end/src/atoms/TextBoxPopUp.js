import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";

import { ChainIconSmallIcon, ExitIcon } from "./icons";

import "./TextBoxPopUp.css";

const MAX_CHAR_FOR_SOURCE_TO_SHOW = 50;

function TextBoxPopUp( {show, text, source,  } ) {

  const [display, setDisplay] = useState(false);

  const handleBackgroundClick = (e) => {
    // clicked outside of modal
    if(e.target.className == "TextBoxPopUp-background") {
      setDisplay(false);
    }
  };

  const handleExitClick = () => {
    setDisplay(false);
  };

  useEffect(() => {
    setDisplay(show);
  }, [show]);

  const handleOnSourceClick = () => {
    alert("Temporary: Link Clicked");
  };

  const renderSource = (srcText) => {
    return srcText.substring(0, MAX_CHAR_FOR_SOURCE_TO_SHOW) + "...";
  };

  return (
    display &&
    <div className="TextBoxPopUp-background" onClick={handleBackgroundClick}>
      <div className="TextBoxPopUp">
        <div className="TextBoxPopUp-exit" onClick={handleExitClick}>
          <ExitIcon/>
        </div>
        <div className="TextBoxPopUp-source" onClick={handleOnSourceClick}>
          <div className="TextBoxPopUp-source-icon">
            <ChainIconSmallIcon/>
          </div>
          <div className="TextBoxPopUp-source-text">
            <a className="TextBoxPopUp-source-text-a"href={source}>{renderSource(source)}</a>
          </div>
        </div>
        <div className="TextBoxPopUp-text">
          {text}
        </div>
      </div>
    </div>
  );
}

TextBoxPopUp.propTypes = {
  text: PropTypes.string.isRequired,
  source: PropTypes.string.isRequired,
  show: PropTypes.bool
};

TextBoxPopUp.defaultProps = {
  show: true
};

export default TextBoxPopUp;