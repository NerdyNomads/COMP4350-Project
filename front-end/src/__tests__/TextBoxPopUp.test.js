/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";

import TextBoxPopUp from "../atoms/TextBoxPopUp";

test("Should match the snapshot.", () => { 
  let testText = "Test";
  let testSource  = "www.abc.com";
  let  onChangeVisibility = jest.fn();
  let wrapper = shallow(<TextBoxPopUp text={testText} source={testSource} onChangeVisibility={onChangeVisibility}/>);
  expect(wrapper.html()).toMatchSnapshot();
});

test("Clicking the background should toggle the pop-up's visibility.", async () => { 
  let testText = "Test";
  let testSource  = "www.abc.com";
  let  onChangeVisibility = jest.fn();
  let wrapper = shallow(<TextBoxPopUp text={testText} source={testSource} onChangeVisibility={onChangeVisibility}/>);
 
  let popUpBackground = wrapper.find(".TextBoxPopUp-background").first();
  expect(popUpBackground.length).toEqual(1);

  let event = {
    target: {
      className: "TextBoxPopUp-background"
    }
  };

  popUpBackground.simulate("click", event);
  expect(onChangeVisibility).toHaveBeenCalledTimes(1);
});

test("Clicking the exit button should toggle the pop-up's visibility.", async () => { 
  let testText = "Test";
  let testSource  = "www.abc.com";
  let  onChangeVisibility = jest.fn();
  let wrapper = shallow(<TextBoxPopUp text={testText} source={testSource} onChangeVisibility={onChangeVisibility}/>);

  let popUpExit = wrapper.find(".TextBoxPopUp-exit").first();
  expect(popUpExit.length).toEqual(1);

  popUpExit.simulate("click");
  expect(onChangeVisibility).toHaveBeenCalledTimes(1);
});
