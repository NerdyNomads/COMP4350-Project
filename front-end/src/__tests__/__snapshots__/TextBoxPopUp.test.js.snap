// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot. 1`] = `"<div class=\\"TextBoxPopUp-background\\"><div class=\\"TextBoxPopUp\\"><div class=\\"TextBoxPopUp-exit\\"><img alt=\\"cross\\" src=\\"/assets/images/delete.svg\\" class=\\"icon-delete\\"/></div><div class=\\"TextBoxPopUp-source\\"><div class=\\"TextBoxPopUp-source-icon\\"><img alt=\\"chain_sm\\" src=\\"/assets/images/chain_sm.svg\\" class=\\"icon\\"/></div><div class=\\"TextBoxPopUp-source-text\\"><a class=\\"TextBoxPopUp-source-text-a\\">www.abc.com...</a></div></div><div class=\\"TextBoxPopUp-text\\">Test</div></div></div>"`;
